rfcs/2022-03-22-motion-rfc.md

```diff
@@ -0,0 +1,611 @@
+- In the above example, we are using `useAnimatedStyle` to create a worklet that will be executed on the UI thread and has a **shared value** `translateXOffset` that is shared by JS as well as native realms.
+- React Native reanimated is able to achieve this using React Native's `TurboModules` feature.
+- Since **we want to leverage these benefits** of **React Native Reanimated**, **we will be using it** for our native **animations on Blade**.
+
+
+# Drawbacks/Constraints
+### Library specific tokens
+Consumers would need to be aware that certain tokens are library specific and would not work with other libraries
+- For web, we expose `duration` as a `string` of *milliseconds* which works well with CSS's `transition-duration` property but will not work with Framer motion since it requires a `number` in *seconds*
+- For web, we expose `easing` as a `string` of `cubic-bezier(..)` which works well with CSS's `transition-timing-function` property but will not work with Framer motion since it requires either,
+  - The name of an existing easing function.
+  - An array of four numbers to define a cubic bezier curve.
+  - An easing function, that accepts and returns a value 0-1.
+- For native, we expose `easing` as a function of `Easing.bezier(..)` imported from `react-native-reanimated` which won't work when used with React Native's `Animated` API.
+
+### React Native `v0.62`
+React Native Reanimated is dependent on `TurboModules`, it restricts us to using React Native `v0.62+` that supports `TurboModules`.
+
+### Multi-step keyframe animation with React Native
+Multi-step animations on web are implemented by having more than 2 steps in an animation. Read more about [multi-step animations here](https://www.joshwcomeau.com/animation/keyframe-animations/#multi-step-animations)
+
+Eg) 2-step keyframe
+```css
+@keyframes move {
+    0% {
+      transform: translateY(0px);
+    }
+    100% {
+      transform: translateY(100px);
+    }
+}
+```
+Eg) Multi-step keyframe
+```css
+@keyframes move {
+    0% {
+      transform: translateY(0px);
+    }
+    50% {
+      transform: translateY(50px);
+    }
+    100% {
+      transform: translateY(100px);
+    }
+}
+```
+
+- In the 2-step keyframe, easing is applied for the animation taking place from 0% to 100%.
+- In the multi-step keyframe, easing is applied for each step of the animation. That is, if we use `ease-in-out` easing, it would be applied to the animation happening from `0%` to `50%` and then again from `50%` to `100%`.
+- This is not supported in React Native Reanimated _directly_ since we apply easing to the `timing` function.
+
```

Comment:
can you attach a side by side example gif to make it visually understandable?

Reply:
This is an edge case where someone would want to replicate the keyframes behavior on native. I've added a link to [this blog](https://www.joshwcomeau.com/animation/keyframe-animations/#multi-step-animations) above that talks more about multi-step animations. Going into examples here would increase the complexity of the RFC.

The intention to keep this documented here is so that _if_ someone wants to do this specific thing in the future, they can refer to this RFC and understand that we were already aware of this edge case and have a workaround described here as well.